Automattic/newspack-network
# feat(content-distribution): prevent older content updating linked posts

By the nature of webhooks, requests might not get processed in order so there's a chance an older version of a distributed post reaching a node. To prevent that from replacing more recent content, we'll use `post_modified_gmt` to bail if the stored payload is more recent.

This PR also changes the post payload date to use `post_date_gmt`. This change allows sites to handle timezones.

### Testing

1. Make sure you have a network setup with at least 1 hub and 1 node
2. On the hub, create a new post, grab the post ID, and use the WP shell to distribute to the node:

```php
$post_id = 0; // Change to your post ID.
$site_urls = [ 'http://node1.local' ]; // Change to match your setup urls.
Newspack_Network\Content_Distribution::set_post_distribution( $post_id, $site_urls );
Newspack_Network\Content_Distribution::distribute_post( $post_id );
```

3. Confirm the post is added to the node
4. Make a change to the post and hit save
5. Confirm the changes are applied to the post in the node
6. Back to the hub shell, manually change some content along with `post_modified` and distribute the post:

```php
$post_id = 0; // Change to your post ID.
$past = '2021-01-01 00:00:00';
wp_update_post( [ 'ID' => $post_id, 'title' => 'Old Title', 'post_modified' => $past, 'post_modified_gmt' => get_gmt_from_date( $past ) ] );
Newspack_Network\Content_Distribution::distribute_post( $post_id );
```

7. Confirm the post does not get updated in the node
